addon/system/store/finders.js

```diff
@@ -56,6 +60,7 @@ export function _findMany(adapter, store, typeClass, ids, internalModels) {
   promise = _guard(promise, _bind(_objectIsAlive, store));
 
   return promise.then(function(adapterPayload) {
+    Ember.assert("You made a `findMany` request for a " + typeClass.typeClassKey + " with ids " + ids + ", but the adapter's response did not have any data", payloadIsNotBlank(adapterPayload));
```

Comment:
@seanpdoyle I would follow the pattern used in this test of stubbing the adapter method to return a value that triggers the assertion then using the `expectAssertion` helper to test that it is called. https://github.com/emberjs/data/blob/1d5dfa142523a4aefdde182262923a42ccb56286/tests/integration/adapter/queries-test.js#L47-L63